test/unit/get.js

```diff
@@ -7,60 +7,76 @@ var manager = require('../manager');
 
 describe('get(session_id, cb)', function() {
 
-	before(manager.setUp);
-	after(manager.tearDown);
-
-	describe('when a session exists', function() {
-
-		var session_id;
-		var data;
-		beforeEach(function(done) {
-			var fixture = _.first(manager.fixtures.sessions);
-			session_id = fixture.session_id;
-			data = fixture.data;
-			manager.populateSession(fixture, done);
-		});
+	_.each([undefined, false, true], function(jsonData) {
 
-		describe('and is not expired', function() {
+		describe('when options.jsonData is set to ' + jsonData, function() {
 
-			it('should return its session data', function(done) {
-				manager.sessionStore.get(session_id, function(error, session) {
-					if (error) return done(error);
-					expect(JSON.stringify(session)).to.equal(JSON.stringify(data));
-					done();
-				});
+			before(manager.tearDown);
+			before(function(done) {
+				manager.createInstance({ jsonData: jsonData }, done);
 			});
-		});
 
-		describe('and is expired', function() {
+			after(manager.tearDown);
 
-			beforeEach(function(done) {
-				manager.expireSession(session_id, done);
-			});
+			describe('when a session exists', function() {
 
-			it('should return NULL', function(done) {
-				manager.sessionStore.get(session_id, function(error, session) {
-					if (error) return done(error);
-					expect(session).to.equal(null);
-					done();
+				var session_id;
+				var data;
+				beforeEach(function(done) {
+					var fixture = _.first(manager.fixtures.sessions);
+					session_id = fixture.session_id;
+					data = fixture.data;
+					manager.populateSession(fixture, done);
 				});
-			});
-		});
-	});
 
-	describe('when a session does not exist', function() {
+				describe('and is not expired', function() {
+
+					it('should return its session data', function(done) {
+						manager.sessionStore.get(session_id, function(error, session) {
+							if (error) return done(error);
+							expect(typeof session).to.equal('object');
+							if (jsonData) {
+								expect(session).to.deep.equal(data);
+							} else {
+								expect(JSON.stringify(session)).to.equal(JSON.stringify(data));
+							}
+							done();
+						});
+					});
+				});
+
+				describe('and is expired', function() {
+
+					beforeEach(function(done) {
+						manager.expireSession(session_id, done);
+					});
+
+					it('should return NULL', function(done) {
+						manager.sessionStore.get(session_id, function(error, session) {
+							if (error) return done(error);
+							expect(session).to.equal(null);
+							done();
+						});
+					});
+				});
+			})
+
+			describe('when a session does not exist', function() {
 
-		beforeEach(manager.clearSessions);
+				beforeEach(manager.clearSessions);
 
-		it('should return NULL', function(done) {
+				it('should return NULL', function(done) {
```

Comment:
I am not sure why GH is highlighting this here. I did try to preserve tab formatting in the file (checked with the linter).  😅